io.typora.Typora.json

```diff
@@ -50,7 +51,11 @@
           "type": "script",
           "dest-filename": "typora",
           "commands": [
-            "exec zypak-wrapper /app/typora/Typora \"$@\""
+            "if [ ! -e $XDG_RUNTIME_DIR/$WAYLAND_DISPLAY ]; then",
```

Comment:
This if statement is always false and break x11 execution.
On x11 `$WAYLAND_DISPLAY` is empty, but the folder `$XDG_RUNTIME_DIR` exists, causing it to always execute with the ozone-platform (and then crash).
Try checking for `$XDG_SESSION_TYPE` instead, or maybe if `$WAYLAND_DISPLAY` is empty or not.
Thanks

Reply:
@syco 
read again. it's not always false. 
the `!` negates the result of `-e $XDG_RUNTIME_DIR/$WAYLAND_DISPLAY`

I tested it by disabling and enabling the wayland socket in flatseal and it switches flawlessly between X11-mode and wayland mode.

It behaves precisely as expected. If the socket does not exist it runs without the extra parameters and therefore in X11 mode. If it exists it runs in wayland mode.